### DISCUSSION TITLE
What does CTNormalization do ?

### DISCUSSION THREAD

**Discussion by sgtm-sougata** [MIC-DKFZ/nnUNet#2363] [Q&A]
We are trying to traina  brain tumor segmentation model on MRI images using nnUNET v2.0. 
When the training starts there is a we can see that the configuration name 3d_lowres using the following parameters:

This is the configuration used by this training:

> Configuration name: 3d_lowres
>  {'data_identifier': 'nnUNetPlans_3d_lowres', 'preprocessor_name': 'DefaultPreprocessor', 'batch_size': 2, 'patch_size': [96, 192, 128], 'median_image_size_in_voxels': [141, 283, 231], 'spacing': [1.1255088100000001, 0.8089594571875001, 0.8089594571875001], 'normalization_schemes': ['CTNormalization'], 'use_mask_for_norm': [False], 'UNet_class_name': 'PlainConvUNet', 'UNet_base_num_features': 32, 'n_conv_per_stage_encoder': [2, 2, 2, 2, 2, 2], 'n_conv_per_stage_decoder': [2, 2, 2, 2, 2], 'num_pool_per_axis': [4, 5, 5], 'pool_op_kernel_sizes': [[1, 1, 1], [2, 2, 2], [2, 2, 2], [2, 2, 2], [2, 2, 2], [1, 2, 2]], 'conv_kernel_sizes': [[3, 3, 3], [3, 3, 3], [3, 3, 3], [3, 3, 3], [3, 3, 3], [3, 3, 3]], 'unet_max_num_features': 320, 'resampling_fn_data': 'resample_data_or_seg_to_shape', 'resampling_fn_seg': 'resample_data_or_seg_to_shape', 'resampling_fn_data_kwargs': {'is_seg': False, 'order': 3, 'order_z': 0, 'force_separate_z': None}, 'resampling_fn_seg_kwargs': {'is_seg': True, 'order': 1, 'order_z': 0, 'force_separate_z': None}, 'resampling_fn_probabilities': 'resample_data_or_seg_to_shape', 'resampling_fn_probabilities_kwargs': {'is_seg': False, 'order': 1, 'order_z': 0, 'force_separate_z': None}, 'batch_dice': False, 'next_stage': '3d_cascade_fullres'} 
>  
> These are the global plan.json settings:
>  {'dataset_name': 'Dataset420_TSCnsGtvp', 'plans_name': 'nnUNetPlans', 'original_median_spacing_after_transp': [4.550000190734863, 0.71875, 0.71875], 'original_median_shape_after_transp': [35, 320, 260], 'image_reader_writer': 'SimpleITKIO', 'transpose_forward': [0, 1, 2], 'transpose_backward': [0, 1, 2], 'experiment_planner_used': 'ExperimentPlanner', 'label_manager': 'LabelManager', 'foreground_intensity_properties_per_channel': {'0': {'max': 8951.0, 'mean': 676.704181320852, 'median': 574.0, 'min': 0.0, 'percentile_00_5': 125.0, 'percentile_99_5': 4021.0, 'std': 502.43467975203697}}} 

We are curious to understand what does normalization schemes: [CTNormalization] mean here.

**Comment by Peaceandmaths** [comment]
From [this](https://arxiv.org/abs/1809.10486) paper : 

" Normalization : Because the intensity scale of CT scans is absolute, all CT
images are automatically normalized based on statistics of the entire respective
dataset: If the modality description in a dataset’s corresponding json desccriptor
file indicates ‘ct’, all intensity values occurring within the segmentation masks
of the training dataset are collected and the entire dataset is normalized by
clipping to the [0.5, 99.5] percentiles of these intensity values, followed by a zscore normalization based on the mean and standard deviation of all collected intensity values. For MRI or other image modalities (i.e. if no ‘ct’ string is
found in the modality), simple z-score normalization is applied to the patient individually. If cropping reduces the average size of patients in a dataset (in voxels) by 1/4 or more the normalization is carried out only within the mask of nonzero
elements and all values outside the mask are set to 0." 

In the source code : https://github.com/MIC-DKFZ/nnUNet/blob/fee8c2db4a52405389eb5d3c4512bd2f654ab999/nnunetv2/preprocessing/normalization/default_normalization_schemes.py#L53
